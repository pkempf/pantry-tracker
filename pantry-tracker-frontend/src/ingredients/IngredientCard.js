import React from "react";
import Card from "react-bootstrap/Card";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Button from "react-bootstrap/Button";

const IngredientCard = ({ ingredient, add, remove }) => {
  return (
    <Card className="IngredientCard mb-2">
      <Card.Body>
        <Card.Title>
          <b>{ingredient.name}</b>
        </Card.Title>
        <Card.Subtitle className="mb-2 text-muted">
          {ingredient.type}
        </Card.Subtitle>
        <Row>
          <Col sm={8}>
            <span className="font-weight-light">
              {ingredient.description || ""}
            </span>
            <span>
              {ingredient.amount ? `Amount: ${ingredient.amount}` : ""}
            </span>
          </Col>
          <Col sm={4}>
            <div className="d-flex justify-content-end m-0 p-0">
              {ingredient.onHand ? (
                <div className="mt-2">
                  <Button variant="dark" disabled>
                    <b>In stock!</b>
                  </Button>
                  <Button variant="danger" className="ml-1" onClick={remove}>
                    (Click to remove)
                  </Button>
                </div>
              ) : (
                <div className="mt-2">
                  <Button variant="secondary" disabled>
                    <b>Out of stock</b>
                  </Button>
                  <Button onClick={add} className="ml-1" variant="success">
                    (Click to add)
                  </Button>
                </div>
              )}
            </div>
          </Col>
        </Row>
      </Card.Body>
    </Card>
  );
};

export default IngredientCard;
