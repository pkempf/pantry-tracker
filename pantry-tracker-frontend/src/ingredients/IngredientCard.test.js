import { render } from "@testing-library/react";
import IngredientCard from "./IngredientCard";

it("renders without crashing", () => {
  render(<IngredientCard ingredient={{}} />);
});

it("matches the snapshot", () => {
  const { asFragment } = render(<IngredientCard ingredient={{}} />);
  expect(asFragment()).toMatchSnapshot();
});
