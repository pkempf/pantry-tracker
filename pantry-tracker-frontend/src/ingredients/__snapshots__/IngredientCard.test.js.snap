// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="IngredientCard mb-2 card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        <b />
      </div>
      <div
        class="mb-2 text-muted card-subtitle h6"
      />
      <div
        class="row"
      >
        <div
          class="col-sm-8"
        >
          <span
            class="font-weight-light"
          />
          <span />
        </div>
        <div
          class="col-sm-4"
        >
          <div
            class="d-flex justify-content-end m-0 p-0"
          >
            <div
              class="mt-2"
            >
              <button
                class="btn btn-secondary"
                disabled=""
                type="button"
              >
                <b>
                  Out of stock
                </b>
              </button>
              <button
                class="ml-1 btn btn-success"
                type="button"
              >
                (Click to add)
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
